XenOvfApi/OVF.cs

```diff
@@ -4307,13 +4307,15 @@ public static VirtualHardwareSection_Type FindVirtualHardwareSectionByAffinity(E
         #endregion
 
         #region SETs
+
         /// <summary>
-        /// Set the count for the number of CPUs to assign to the virtual machine.
+        /// Set the count for the number of vCPUs and max vCPUs to assign to the virtual machine.
         /// </summary>
         /// <param name="ovfObj">EnvelopeType</param>
         /// <param name="vsId">Virtual System Id</param>
-        /// <param name="cpucount">Number of CPUs</param>
-		public static void SetCPUs(EnvelopeType ovfEnv, string vsId, UInt64 cpucount)
+        /// <param name="cpucount">Number of vCPUs</param>
+        /// <param name="maxCpuCount">Number of max vCPUs, if available</param>
+        public static void SetCPUs(EnvelopeType ovfEnv, string vsId, UInt64 cpucount, ulong? maxCpuCount = null)

```

Comment:
Instead of making this nullable, you can define it as `ulong maxCpuCount = 100_000` and then further down it can be just `rasd.Limit = new cimUnsignedLong { Value = maxCpuCount };`